### PR TITLE
fix: add integration tests

### PR DESCRIPTION
**Pull request by GuyAb**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔/✖                                                                        |
| New feature     | ✔/✖                                                                        |
| Breaking change | ✔/✖                                                                        |
| Deprecations    | ✔/✖                                                                        |
| Documentation   | ✔/✖                                                                        |
| Tests added     | ✔/✖                                                                        |

Related issues: #XXX , #XXX ...
Closes #XXX , #XXX ...

Further  information:

### CODE REVIEW

File: tests/integration-tests/jobs/createJobKubernetes-test.js [hunk 1]
```diff
@@ -218,12 +224,150 @@ describe('Create job specific kubernetes tests', async function () {
...
+                    });
+
+                    it('Create job which is one time load_test', async () => {
+                        nock(kubernetesConfig.kubernetesUrl).post(`/apis/batch/v1/namespaces/${kubernetesConfig.kubernetesNamespace}/jobs`)
+                            .reply(200, {
+                                metadata: { name: 'jobName', uid: 'uid' },
+                                namespace: kubernetesConfig.kubernetesNamespace
+                            });
+
+                        const jobBody = {
```
Comment (kerenfi) [inline tests/integration-tests/jobs/createJobKubernetes-test.js:264]: let's put job body as const outside the test scope - just to similify the test scope length - can be a function with params

File: tests/integration-tests/jobs/createJobKubernetes-test.js [hunk 2]
```diff
@@ -218,12 +224,150 @@ describe('Create job specific kubernetes tests', async function () {
...
+                        createJobResponse = await schedulerRequestCreator.createJob(jobBody, {
+                            'Content-Type': 'application/json'
+                        });
+
+                        should(createJobResponse.status).eql(201);
+                        jobId = createJobResponse.body.id;
+                    });
+
+                    it('use delete all containers', async () => {
+                        nock(kubernetesConfig.kubernetesUrl).get(`/apis/batch/v1/namespaces/${kubernetesConfig.kubernetesNamespace}/jobs?labelSelector=app=predator`)
```
Comment (kerenfi) [inline tests/integration-tests/jobs/createJobKubernetes-test.js:292]: lots of nocks preparation - do we want it in the test scope?

File: tests/integration-tests/jobs/createJobKubernetes-test.js [hunk 3]
```diff
@@ -218,12 +224,150 @@ describe('Create job specific kubernetes tests', async function () {
...
+
+                        chaosExperimentsRequestCreator.nockK8sChaosExperimentSupportedKinds();
+
+                        nock(kubernetesConfig.kubernetesUrl).get('/apis/chaos-mesh.org/v1alpha1/podchaos?labelSelector=app=predator')
+                            .reply(200, {
+                                items: [{
+                                    metadata: { name: 'first-exp', namespace: kubernetesConfig.kubernetesNamespace },
+                                    status: {
+                                        experiment: {
+                                            desiredPhase: 'Stop'
```
Comment (kerenfi) [inline tests/integration-tests/jobs/createJobKubernetes-test.js:327]: do we test that desiredPhase != stop is not deleted?

do we test failure scenarios - k8s api fails to delete

do we test non app=predator label not deleted (?)

Comment (GuyAb) [reply tests/integration-tests/jobs/createJobKubernetes-test.js:327]: 1. yes - you can see that I used one with Run and it was not counted as deleted in counter - only 2 deleted.
2. No, but I think this whole scope doesn't check failures from kubernetes, we check it on unit tests - here the purpose is to check the flow.
3. Not really, but we only do get for predator label so we don't really have that info.

Comment (kerenfi) [reply tests/integration-tests/jobs/createJobKubernetes-test.js:327]: 1 - resolve
2- okish even though it is possible to test
3 resolve

File: tests/integration-tests/jobs/createJobKubernetes-test.js (hunk 1 above)
Comment (kerenfi) [reply tests/integration-tests/jobs/createJobKubernetes-test.js:264]: is it possible here to take it out of the scope?

File: tests/integration-tests/jobs/createJobKubernetes-test.js (hunk 3 above)
Comment (GuyAb) [reply tests/integration-tests/jobs/createJobKubernetes-test.js:327]: For 2 we will have a separated PR for these negative scenarios

**Pull request merged**